File: app/controllers/contacts_controller.rb
class ContactsController < ApplicationController
  def index
    if params[:search]
      contacts = Contact.where('name ILIKE ? OR number ILIKE ?',
                               "%#{params[:search]}%",
                               "%#{params[:search]}%")

      render json: { contacts: } and return
    end

    @contacts = Contact.select(:name, :number)
  end

  def show

  end
end

File: app/javascript/controllers/contacts_controller.js
import { Controller } from "@hotwired/stimulus"

// Connects to data-controller="contacts"
export default class extends Controller {
  static targets = ["search", "searchTarget", "contactItem"]

  connect() {

  }

  searchContacts(){
    let lowerCaseSearchInput = this.searchTarget.value.toLowerCase()

    this.contactItemTargets.forEach((element, index) => {
      const searchValue = element.getAttribute('data-search-key')

      element.style.display = searchValue.includes(lowerCaseSearchInput)
                            ? "block"
                            : "none"
    })
  }

  openContactShow(event){
    const contactInfo = event.currentTarget.getAttribute('data-contact-show-params')
    // const modal = this.modalTarget
    const contactJSON = JSON.parse(contactInfo)
    console.log(contactJSON)
    // modal.style.display = 'block'
    // modal.innerHTML = contactJSON
  }

  cloaseContactShow(){
    modal.style.display = 'none'
  }
}

